src/librbd/librbd.cc

```diff
@@ -2698,20 +2701,37 @@ namespace librbd {
   int Image::diff_iterate2(const char *fromsnapname, uint64_t ofs, uint64_t len,
                            bool include_parent, bool whole_object,
                            int (*cb)(uint64_t, size_t, int, void *), void *arg)
-  {
+  {  
     ImageCtx *ictx = (ImageCtx *)ctx;
     tracepoint(librbd, diff_iterate_enter, ictx, ictx->name.c_str(),
               ictx->snap_name.c_str(), ictx->read_only, fromsnapname, ofs, len,
               include_parent, whole_object);
+    cls::rbd::SnapshotNamespace snapshot_namespace = {
```

Comment:
No need to define `cls::rbd::SnapshotNamespace` variables -- concrete namespace can be constructed in place: `ictx->get_snap_id(cls::rbd::UserSnapshotNamespace(), ...)`